src/webgpu/api/validation/buffer/mapping.spec.ts

```diff
@@ -311,10 +335,42 @@ g.test('mapAsync,earlyRejection')
 
     const buffer = t.createMappableBuffer(mapMode, bufferSize);
     const p1 = buffer.mapAsync(mapMode, offset1, mapSize); // succeeds
-    await t.testMapAsyncCall(false, true, 'OperationError', buffer, mapMode, offset2, mapSize);
+    await t.testMapAsyncCall(
+      false,
+      false,
+      true,
+      'OperationError',
+      buffer,
+      mapMode,
+      offset2,
+      mapSize
+    );
     await p1; // ensure the original map still succeeds
   });
 
+g.test('mapAsync,abort_over_invalid_error')
+  .desc('Test that unmap abort error should have precedence over validation error')
+  .paramsSubcasesOnly(u =>
+    u.combine('mapMode', kMapModeOptions).combine('abortError', [true, false])
+  )
+  .fn(async t => {
+    const { mapMode, abortError } = t.params;
+    const bufferSize = 8;
+    const buffer = t.createMappableBuffer(mapMode, bufferSize);
+    await buffer.mapAsync(mapMode);
+
+    if (abortError) {
+      // unmap abort error should have precedence over validation error
+      const pending = t.testMapAsyncCall(false, false, false, 'AbortError', buffer, mapMode);
+      buffer.unmap();
+      await pending;
+    } else {
+      // map on already mapped buffer should cause validation error
+      await t.testMapAsyncCall(false, true, false, 'OperationError', buffer, mapMode);
+      buffer.unmap();
+    }
+  });
```

Comment:
Do you think we should test that unmap should have precedence over not only "map on already mapped buffer" but also other invalidation errors eg: "map async request on invalid buffer", "offset is not a multiple of 8"? Or does it sound duplicated?

Reply:
I think we should. It's reasonable for implementation to lift some validation to the content process, for example "offset is not a multiple of 8" can be validated easily in the content process, whereas "invalid buffer" is impossible to validate in the content process.

Reply:
> I think we should. 

OK, I added a TODO for now.